### ISSUE TITLE
Help remove git dependency from jingo-minify in order to reduce container image size

### ISSUE THREAD

**Issue by bqbn** [mozilla/addons-server#3503]
### What did you do?

Build a container image using Dockerfile.deploy file
### What happened?

The latest AMO container image is about 1.7GB in size, one third of which is due to the jingo-minify module having a one line dependency on the git history, and as a result, we have to copy the entire .git/ directory (about 600MB at the moment) into the image.

https://github.com/yohanboniface/jingo-minify/blob/master/jingo_minify/management/commands/compress_assets.py#L42
### What did you expect to happen?

Remove this run time dependency on git from jingo-minify so we have a smaller sized container image.
### Anything else we should know?

One idea mentioned in email thread is to leverage the version.json file to obtain git commit number, which implies that the build system needs to inject the version.json prior to running "docker build". Also, not sure if using a UUID to replace the git commit hash is a good idea in this case.

**Comment by diox** [comment]
As a quick workaround, I mentioned doing a shallow clone (with `--depth=1`) to avoid loading all the history. Any thoughts about that ?

**Comment by bqbn** [comment]
Ah, did you reply to the email thread or mention it somewhere else previously?

The idea may work. I'll try it out in -dev and -stage this week and report back.

**Comment by muffinresearch** [comment]
@bqbn how did this work out - did using a shallow clone provide enough of a reduction or do we need to look at other changes?

**Comment by bqbn** [comment]
Using shallow clone was able to reduce the image size to a little over 1 GB. The .git/ directory is ~50 MB. I think this lowers the urgency of this issue for us.

Do you guys still want to try to remove the run time dependency on git from jingo-minify? One potential concern of having this dependency is the .git/ can get bigger if there are too many static files in the repo in the future.

**Comment by muffinresearch** [comment]
@bqbn Whilst it's not going to happen short-term we're like to get rid of this eventually as we move towards using addons-frontend to replace the addons-server frontend. 

If you're ok for now then I think we can close this on the basis that if it becomes an issue again in the future we can reopen it.

**Comment by bqbn** [comment]
Sounds good to me. Will close this for now.

**Comment by bqbn** [comment]
I'd like to re-open this issue for consideration. 

The tld'r is that one of our dependencies, namely jingo_minify, has a one line dependency on the git history, 

https://github.com/yohanboniface/jingo-minify/blob/master/jingo_minify/management/commands/compress_assets.py#L42

and as a result, we have to copy the entire .git/ directory (about 833MB at the moment) into the container image.

Previously to work around the issue, we shallow-cloned the addons-server repo with a depth of one. However, since we moved from our own Jenkins build server to CircleCI, we have lost the ability to do shallow clones because CircleCI doesn't provide an option for us to do so (CircleCI needs to read the circle.yml file first, and in order to get that file, it checks out the whole repository).

Long story short, our container image size gets big again. 

Having a smaller size container image helps reduce build and deployment time. Please re-consider. Thanks! :)

**Comment by diox** [comment]
Yeah. We want to get rid of jingo-minify but that is unlikely to happen soon... We'll look into just generating a UUID instead of using the git commit id...

**Comment by diox** [comment]
I added a `--use-uuid` option to `compress_assets` command and upgraded jingo-minify in #5058. I'll follow up with a PR to use that option.
